app/services/register_and_partner_api/sync_users.rb

```diff
@@ -38,9 +39,40 @@ def self.user_attributes_from(user_from_api)
         register_and_partner_id: user_from_api.attributes[:id],
         full_name: user_from_api.attributes[:attributes][:full_name],
         email: user_from_api.attributes[:attributes][:email],
-        cip: "UCL",
+        user_type: user_from_api.attributes[:attributes][:user_type],
+        cip: user_from_api.attributes[:attributes][:core_induction_programme],
       }
     end
-    private_class_method :sync_users, :user_attributes_from
+
+    def self.set_core_induction_programme(attributes, profile)
+      profile.core_induction_programme = CoreInductionProgramme.find_by(
+        name: determine_core_induction_programme(attributes[:cip]),
+      )
+    end
+
+    def self.determine_core_induction_programme(cip_name)
+      case cip_name
+      when "ambition"
+        "Ambition Institute"
+      when "edt"
+        "Education Development Trust"
+      when "teach_first"
+        "Teach First"
+      when "ucl"
+        "UCL"
+      end
+    end
+
+    def self.set_profile_type(user, user_type)
+      case user_type
+      when "early_career_teacher"
+        ::EarlyCareerTeacherProfile.find_or_initialize_by(user: user)
+      when "other"
+        ::MentorProfile.find_or_initialize_by(user: user)
+      end
```

Comment:
I would say when `other`, we shouldn't do anything at all. 

Maybe add `mentor` type just on our side, and make a mentor profile for them, but for `other` user type we should do nothing, not even create a user (I think).

Reply:
It doesn't seem like not having something assigned to profile before calling save on user. 

If we call user.save! before assigning a profile it works. Unless there
The below works when we don't create a mentor

`user.save!
profile = find_or_initialize_profile(user, attributes[:user_type])
set_core_induction_programme(attributes, profile)
profile.save!`